bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/handler/TuyaDeviceHandler.java

```diff
@@ -0,0 +1,658 @@
+            if (command instanceof StringType) {
+                switch (configuration.irType) {
+                    case "base64" -> {
+                        commandRequest.put(1, "study_key");
+                        commandRequest.put(7, command.toString());
+                    }
+                    case "tuya-head" -> {
+                        if (!configuration.irCode.isBlank()) {
+                            commandRequest.put(1, "send_ir");
+                            commandRequest.put(3, configuration.irCode);
+                            commandRequest.put(4, command.toString());
+                            commandRequest.put(10, configuration.irSendDelay);
+                            commandRequest.put(13, configuration.irCodeType);
+                        } else {
+                            logger.warn("irCode is not set for channel {}", channelUID);
+                        }
+                    }
+                    case "nec" -> {
+                        long code = convertHexCode(command.toString());
+                        String base64Code = IrUtils.necToBase64(code);
+                        commandRequest.put(1, "study_key");
+                        commandRequest.put(7, base64Code);
+                    }
+                    case "samsung" -> {
+                        long code = convertHexCode(command.toString());
+                        String base64Code = IrUtils.samsungToBase64(code);
+                        commandRequest.put(1, "study_key");
+                        commandRequest.put(7, base64Code);
+                    }
+                }
+                irStopLearning();
+            }
+        }
+
+        TuyaDevice tuyaDevice = this.tuyaDevice;
+        if (!commandRequest.isEmpty() && tuyaDevice != null) {
+            tuyaDevice.set(commandRequest);
+        }
+
+        if (CHANNEL_TYPE_UID_IR_CODE.equals(channelTypeUID)) {
+            if (command instanceof StringType) {
+                irStartLearning(configuration.activeListen);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        disposing = true;
+        ScheduledFuture<?> future = reconnectFuture;
+        if (future != null) {
+            future.cancel(true);
+        }
+        future = this.pollingJob;
+        if (future != null) {
+            future.cancel(true);
+        }
+        if (configuration.ip.isEmpty()) {
+            // unregister listener only if IP is not fixed
+            udpDiscoveryListener.unregisterListener(this);
+        }
+        TuyaDevice tuyaDevice = this.tuyaDevice;
+        if (tuyaDevice != null) {
+            tuyaDevice.dispose();
+            this.tuyaDevice = null;
+        }
+        irStopLearning();
+    }
+
+    @Override
+    public void initialize() {
+        // clear all maps
+        dpToChannelId.clear();
+        dp2ToChannelId.clear();
+        channelIdToChannelTypeUID.clear();
+        channelIdToConfiguration.clear();
```

Comment:
I guess this can safely be moved to `dispose`?

Reply:
Probably it would be better to create new maps on `initialize`, just to be sure.